Execution scope clarification (#20269)
microsoftgraph/microsoft-graph-docs: api-reference/beta/api/workflow-list-executionscope.md

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> List the users that meet the [execution conditions](../resources/identitygovernance-workflowexecutionconditions.md) regardless of whether they have already been processed by the [workflow](../resources/identitygovernance-workflow.md).

Reviewer: Per the guidance in the Microsoft Writing Style Guide, we use alternate words instead of variations of "execute", even if it is used in this case in the resource name. I suggest revising "execution conditions" to "run conditions", for globalization/localization. Please revise occurrences throughout.

Author: @Lauragra I hear you but then we have a process issue, it should not have been approved in Graph API review, now the ship has pretty much sailed for this. Can we merge this PR and then start a separate conversation about it? We have lots of docs pubic already that speak about execution conditions, so it's not isolated to this PR.

Reviewer: @AlexFilipin , okay, to be clear, are you referring to the API review of the resource names? We can also take up that conversation, but even if we don't rename the APIs, we can adjust the words we use in the content. If you want to do a separate PR to globally update the text, then I can merge and publish this PR for now.

Author: Yes, I am referring to the API review of the resource names / APIs. I think it's tricky to use another terminology in the content but then cross-link to a resource with a different name. I would like to merge this PR as is and start a separate conversation if we have to update these resource names / APIs.

E.g., I could update this PR to speak about run conditions but then it cross-links to a resource called execution conditions (that would be somewhat weird).

Our feature and API docs are full of "execution conditions" we use it so often and much I am surprised it is coming up only now, but I understand the concern.

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

List the users that meet the [execution conditions](../resources/identitygovernance-workflowexecutionconditions.md) regardless of whether they have already been processed by the [workflow](../resources/identitygovernance-workflow.md).

> The users are periodically evaluated and not determined in real time in the course of the API call, there will be delays until the result is accurate if the execution conditions have been changed recently, relevant attributes on the user have been changed recently or the [time based trigger](../resources/identitygovernance-timebasedattributetrigger.md) has been reached recently.

Reviewer 2: ```suggestion
The workflow engine periodically evaluates the users that meet the execution conditions. The results will not be up to date if the execution conditions have been changed recently, relevant attributes on the user have been changed recently or the [time based trigger](../resources/identitygovernance-timebasedattributetrigger.md) has been reached recently.
```

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

List the users that meet the [execution conditions](../resources/identitygovernance-workflowexecutionconditions.md) regardless of whether they have already been processed by the [workflow](../resources/identitygovernance-workflow.md).

The users are periodically evaluated and not determined in real time in the course of the API call, there will be delays until the result is accurate if the execution conditions have been changed recently, relevant attributes on the user have been changed recently or the [time based trigger](../resources/identitygovernance-timebasedattributetrigger.md) has been reached recently.

> Since the [time based trigger](../resources/identitygovernance-timebasedattributetrigger.md) is a time window that looks back three days, the result of this API call will also return users whose execution day is slightly in the past.

Reviewer 2: Hmm, I'm not getting this part; specifically:
+ ".. there will be delays until the result is accurate" - This sounds to me like the API is a long-running operation; which it doesn't seem to be.
+ "Since the [time based trigger](../resources/identitygovernance-timebasedattributetrigger.md) is a time window that looks back three days" - I'm missing where the three days is coming from. Not seeing it in the docs.

Let's sync internally to help me understand

> No results are retruned for workflows if the schedule is disabled.
> 
> No results are retruned for workflows with [on-demand execution conditions](../resources/identitygovernance-ondemandexecutiononly.md).
> 

Reviewer 2: Does "workflows if the schedule is disabled" mean isEnabled is `false`? Then consider the suggested edits.
```suggestion
No results are returned for disabled workflows or for workflows with [on-demand execution conditions](../resources/identitygovernance-ondemandexecutiononly.md).

```
